server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -116,19 +116,64 @@ void afterEach() {
     class ValidationTest {
 
         @Test
-        void restoreShouldReturnInvalidWhenUserIsInvalid() {
+        void restoreShouldReturnInvalidWhenActionIsMissing() {
             when()
+                .post(USER.asString())
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("action cannot be null, empty or blank"));
+        }
+
+        @Test
+        void restoreShouldReturnInvalidWhenActionIsInValid() {
+            given()
+                .queryParam("action", "invalid action")
+            .when()
+                .post(USER.asString())
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("'invalid action' is not a valid action. Supported values are: (restore)"));
+        }
+
+        @Test
+        void restoreShouldReturnInvalidWhenPassingCaseInsensitiveAction() {
+            given()
+                .queryParam("action", "RESTORE")
+            .when()
+                .post(USER.asString())
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("'RESTORE' is not a valid action. Supported values are: (restore)"));
+        }
+
+        @Test
+        void restoreShouldReturnInvalidWhenUserIsInvalid() {
+            given()
+                .queryParam("action", "restore")
+            .when()
                 .post("not@valid@user.com")
             .then()
                 .statusCode(HttpStatus.BAD_REQUEST_400)
                 .body("statusCode", is(400))
                 .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
-                .body("message", is("The username should not contain multiple domain delimiter."));
+                .body("message", is("Invalid arguments supplied in the user request"))
+                .body("details", is("The username should not contain multiple domain delimiter."));
```

Comment:
having several `@` is actually valid AFAIK

Reply:
`User.fromUsername(String)` throws that message. Do you want to revise that method?

Reply:
Ok, it's ok for a mail address (because it's what RFCs define) but for a User we can decide what is valid. You can keep it as-is.